### PR TITLE
Handle missing variable declaration in foreach loop in VariablesDeclared

### LINKED ISSUE #17971: Extract method refactoring exception for incomplete foreach
**Version Used**: VS 2017, March 14, 2017 (build 26228.09)

**Steps to Reproduce**:

1.  Create .NET 4.6.1 console project.
2. Enter this code in Program.cs:
```
using System;
namespace ConsoleApp1
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("1"); // select from here
            foreach () // can be without braces - result is same
            Console.WriteLine("2"); // to here
        }
    }
}
```
3. Select all code in Main method.

**Expected Behavior**:
Lightbulb with Extract Method refactoring appears in editor or no refactorings appears (because code is broken).

**Actual Behavior**:
`'ExtractMethodCodeRefactoringProvider' encountered an error and has been disabled` appears at top of editor with stack trace:
```
System.InvalidOperationException : Sequence contains no elements
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)
   at Microsoft.CodeAnalysis.CSharp.VariablesDeclaredWalker.VisitForEachIterationVariables(BoundForEachStatement node)
   at Microsoft.CodeAnalysis.CSharp.PreciseAbstractFlowPass`1.VisitForEachStatement(BoundForEachStatement node)
   at Microsoft.CodeAnalysis.CSharp.BoundForEachStatement.Accept(BoundTreeVisitor visitor)
   at Microsoft.CodeAnalysis.CSharp.PreciseAbstractFlowPass`1.VisitWithStackGuard(BoundNode node)
   at Microsoft.CodeAnalysis.CSharp.PreciseAbstractFlowPass`1.VisitAlways(BoundNode node)
   at Microsoft.CodeAnalysis.CSharp.AbstractRegionControlFlowPass.Visit(BoundNode node)
   at Microsoft.CodeAnalysis.CSharp.ControlFlowPass.VisitStatement(BoundStatement statement)
   at Microsoft.CodeAnalysis.CSharp.PreciseAbstractFlowPass`1.VisitStatements(ImmutableArray`1 statements)
   at Microsoft.CodeAnalysis.CSharp.PreciseAbstractFlowPass`1.VisitBlock(BoundBlock node)
   at Microsoft.CodeAnalysis.CSharp.BoundBlock.Accept(BoundTreeVisitor visitor)
   at Microsoft.CodeAnalysis.CSharp.PreciseAbstractFlowPass`1.VisitWithStackGuard(BoundNode node)
   at Microsoft.CodeAnalysis.CSharp.PreciseAbstractFlowPass`1.VisitAlways(BoundNode node)
   at Microsoft.CodeAnalysis.CSharp.AbstractRegionControlFlowPass.Visit(BoundNode node)
   at Microsoft.CodeAnalysis.CSharp.PreciseAbstractFlowPass`1.Scan(Boolean& badRegion)
   at Microsoft.CodeAnalysis.CSharp.ControlFlowPass.Scan(Boolean& badRegion)
   at Microsoft.CodeAnalysis.CSharp.PreciseAbstractFlowPass`1.Analyze(Boolean& badRegion)
   at Microsoft.CodeAnalysis.CSharp.VariablesDeclaredWalker.Analyze(CSharpCompilation compilation,Symbol member,BoundNode node,BoundNode firstInRegion,BoundNode lastInRegion)
   at Microsoft.CodeAnalysis.CSharp.CSharpDataFlowAnalysis.get_VariablesDeclared()
   at Microsoft.CodeAnalysis.ExtractMethod.MethodExtractor.Analyzer.GenerateVariableInfoMap(SemanticModel model,DataFlowAnalysis dataFlowAnalysisData,Dictionary`2 symbolMap)
   at async Microsoft.CodeAnalysis.ExtractMethod.MethodExtractor.Analyzer.AnalyzeAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.ExtractMethod.MethodExtractor.ExtractMethodAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.ExtractMethod.AbstractExtractMethodService`3.ExtractMethodAsync[TValidator,TExtractor,TResult](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.ExtractMethod.ExtractMethodCodeRefactoringProvider.GetCodeActionAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.ExtractMethod.ExtractMethodCodeRefactoringProvider.ComputeRefactoringsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync(<Unknown Parameters>)
```

### PR DESCRIPTION
**Pull request by gafter**
Fixes #17971 

@dotnet/roslyn-compiler Please review this small bug fix.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/FlowAnalysis/VariablesDeclaredWalker.cs [hunk 1]
```diff
@@ -131,12 +131,11 @@ public override void VisitForEachIterationVariables(BoundForEachStatement node)
 
                 if (deconstructionAssignment == null)
                 {
-                    // regular, not deconstructing, foreach declares exactly one iteration variable
-                    _variablesDeclared.Add(node.IterationVariables.Single());
+                    _variablesDeclared.AddAll(node.IterationVariables);
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/FlowAnalysis/VariablesDeclaredWalker.cs]: >_variablesDeclared.AddAll(node.IterationVariables); [](start = 20, length = 51)

Would it be possible to add a targeted compiler test for the affected Region Analysis API?

**Comment by AlekseyTs** [comment]
Done with review pass.

File: src/Compilers/CSharp/Portable/FlowAnalysis/VariablesDeclaredWalker.cs (hunk 1 above)
Comment (gafter) [inline src/Compilers/CSharp/Portable/FlowAnalysis/VariablesDeclaredWalker.cs]: Yes, done.

**Comment by gafter** [comment]
@dotnet/roslyn-compiler May I please have a second review?

**Comment by gafter** [comment]
@dotnet/roslyn-compiler May I please have a second review?

**Comment by gafter** [comment]
@dotnet/roslyn-compiler May I please have a second review? @AlekseyTs If you're happy with this now, would you please approve?

**Comment by gafter** [comment]
@jaredpar Do I need approval to merge?

**Comment by jaredpar** [comment]
Approved for merge

**Pull request merged**